### PR TITLE
Clean up typing-related imports in _dihttp.py

### PR DESCRIPTION
**Pull request by wsanchez**
This cleans up typing-related imports in `_dihttp.py`.

It also ratchets up the `mypy` config a bit.

### CODE REVIEW

**Comment by twm** [comment]
I don't understand what's going on here... the title says "add type hints" but this doesn't appear to add any hints?

**Comment by wsanchez** [comment]
@twm Whoops, true.  I reused the same ticket name for each file as I peeled away bits from #301 into smaller PRs in hopes of enabling quicker reviews.

This PR is really just cleanup.  Title and description updated.

File: tox.ini [hunk 1]
```diff
@@ -283,6 +283,7 @@ commands =
 
 # Global settings
 
+disallow_incomplete_defs = True
```
Comment (wsanchez) [inline tox.ini:286]: No incomplete definitions left; we can enable this check now.

File: tox.ini [hunk 2]
```diff
@@ -304,9 +304,6 @@ warn_return_any          = False
 [mypy-klein._app]
 allow_untyped_defs = True
 
-[mypy-klein._decorators]
-allow_untyped_defs = True
-
```
Comment (wsanchez) [inline tox.ini:309]: This file is clean now; no need for the exception.

File: tox.ini (hunk 1 above)
Comment (twm) [reply tox.ini:282]: :tada:

**Pull request merged**